Profiles for measurement area

schroedtert opened the pull request:
Introduces the possibility to compute profiles for a specified axis-aligned measurement area.

For testing this: 

- Install PedPy from the branch:
```
pip install --force-reinstall git+https://github.com/schroedtert/PedPy.git@profiles-for-measurement-area
```

- Take a look at the attached notebook: 
[profiles.ipynb.zip](https://github.com/user-attachments/files/21394387/profiles.ipynb.zip)


Rough overview:

```python
# create axis aligned measurement area
measurement_area = AxisAlignedMeasurmentArea(-1, 0.5, 1, 1.3)

# usual precomputations

# compute profile in measurement area
voronoi_density_profile = compute_density_profile(
    data=resorted_profile_data,
    axis_aligned_measurement_area=measurement_area,
    grid_intersections_area=grid_cell_intersection_area,
    grid_size=grid_size,
    density_method=DensityMethod.VORONOI,
)

# plot result
plot_profiles(
    walkable_area=walkable_area,
    measurement_area=measurement_area,
    profiles=voronoi_density_profile,
    axes=ax0,
    label="$\\rho$ / 1/$m^2$",
    vmin=0,
    vmax=8,
    title="Voronoi",
)

```

<img width="1214" height="511" alt="image" src="https://github.com/user-attachments/assets/e106f9c4-2fff-4008-91c0-07f47e9121af" />

Closes #413

JuleAdrian commented:
There is a small error in the provided profiles.ipynb:
In the notebook you wrote `measurement_area = AxisAlignedMeasurmentArea((-1, 0.5, 1, 1.3))`, 
but it should rather be `measurement_area = AxisAlignedMeasurmentArea(-1, 0.5, 1, 1.3)` 
or `measurement_area = AxisAlignedMeasurmentArea(x_min = -1,  y_min = 0.5,  x_max = 1,  y_max = 1.3) `.

However, this has nothing to do with the code..

schroedtert commented on pyproject.toml:
I stumbled upon it on my machine and wanted to fix it directly. This is not directly related to this feature, hence there is the second commit in the history. So when we rebase this PR to main, both commits will be in the history. If you prefer, I can also move this to a separate PR.

JuleAdrian replied:
No, it's fine like this. I just wanted to understand. Good, that it appears as separate commit in the history!

schroedtert commented:
> * Include the following line in user_guide: aa_ma = AxisAlignedMa()ma.bounds

Slightly adapted the approach, added a new function `AxisAlignedMeasurementArea.from_measurement_area(measurement_area)` for this purpose.

>     * check if the input measurement area needs to be an AxisAlignedMa, or if also a usual ma is accepted by the function

Added the check, with hint to use the `from_measurement_area` function.

>     * in the user guide: use a aa_ma that fits well to the grid so that the results are as similar as possible

Changed the measurement area, how it looks quite good I would say:

<img width="1214" height="511" alt="image" src="https://github.com/user-attachments/assets/e0a6804d-291d-45e8-a909-77a45de45dd8" />

<img width="1214" height="511" alt="image" src="https://github.com/user-attachments/assets/7d97634e-4714-4abe-b9e7-5cf1b4b6f54d" />
